pkg/controller/node/node_controller.go

```diff
@@ -1022,13 +1036,50 @@ func (ctrl *Controller) setUpdateInProgressTaint(ctx context.Context, nodeName s
 			newNode.Spec.Taints = []corev1.Taint{}
 		}
 
+		if checkIfNodeHasInProgressTaint(newNode) {
```

Comment:
Should we also move this check out? i.e. in the sync code, have:

```
if node.Annotations[daemonconsts.DesiredMachineConfigAnnotationKey] != targetConfig && !checkIfNodeHasInProgressTaint(node)
```

just to keep the two functions consistent

Reply:
Yup we can do that.

Reply:
Done.